pygmt/tests/test_grdfill.py

```diff
@@ -0,0 +1,44 @@
+"""
+Tests for grdclip.
+"""
+import os
+
+import numpy as np
+import pytest
+import xarray as xr
+from pygmt import grdfill, grdinfo
+from pygmt.datasets import load_earth_relief
+from pygmt.helpers import GMTTempFile
+
+
+@pytest.fixture(scope="module", name="grid")
+def fixture_grid():
+    """
+    Load the grid data from the sample earth_relief file and set value(s) to
+    NaN.
+    """
+    grid = load_earth_relief(registration="pixel")
+    grid[10, 10] = np.nan
```

Comment:
Prefer to use a smaller 10x10 region instead of a 180x360 pixel grid, and set more than one pixel to NaN.

```suggestion
    grid = load_earth_relief(registration="pixel", region=[-5, 5, -5, 5])
    grid[3:5, 3:5] = np.nan
    grid[5:7, 5:7] = np.inf
```

I've also tried setting a 2x2 group of pixels to [`np.inf`](https://numpy.org/devdocs/reference/constants.html#numpy.inf) (infinity), and these don't seem to be filled in by `grdfill`, which is to be expected I think?